<!--
    ROS navigation stack with velocity smoother and safety (reactive) controller
-->
<launch>
  <arg name="odom_frame_id"   default="odom"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="odom_topic" default="odom" />
  <arg name="laser_topic" default="scan" />
  <arg name="custom_param_file" default="$(find sarl_star_ros)/param/dummy.yaml"/>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find sarl_star_ros)/param/config/odom/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find sarl_star_ros)/param/config/odom/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find sarl_star_ros)/param/config/odom/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find sarl_star_ros)/param/config/odom/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find sarl_star_ros)/param/config/odom/base_local_planner_params.yaml" command="load" />
    <rosparam file="$(find sarl_star_ros)/param/dwa_local_planner_params.yaml" command="load" />
    <rosparam file="$(find sarl_star_ros)/param/config/nav_obstacles_params.yaml" command="load" />
    <!--<rosparam file="$(find turtlebot_navigation)/param/dwa_local_planner_params.yaml" command="load" />
    <rosparam file="$(find turtlebot_navigation)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find turtlebot_navigation)/param/global_planner_params.yaml" command="load" />
    <rosparam file="$(find turtlebot_navigation)/param/navfn_global_planner_params.yaml" command="load" />-->
    <!-- external params file that could be loaded into the move_base namespace
    <rosparam file="$(arg custom_param_file)" command="load" />-->

    <!-- reset frame_id parameters using user input data -->
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>

    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
  </node>
</launch>

